kuma/users/tests/test_views.py

```diff
@@ -1295,26 +1297,203 @@ def test_account_refresh_token_saved_next_login(self):
+    url = reverse('users.user_delete', kwargs={'username': wiki_user.username})
+    response = user_client.get(url, HTTP_HOST=settings.WIKI_HOST)
+    assert response.status_code == 200
+    response = user_client.post(url, HTTP_HOST=settings.WIKI_HOST)
+    assert response.status_code == 302
+    assert not User.objects.filter(username=wiki_user.username).exists()
+
+
+def test_delete_user_donate_attributions(db, user_client, wiki_user):
+
+    # Pretend the user logged in with GitHub
+    SocialAccount.objects.create(
+        user=wiki_user,
+        provider='github',
+        extra_data=dict(
+            email=wiki_user.email,
+            avatar_url='https://avatars0.githubusercontent.com/yada/yada',
+            html_url="https://github.com/{}".format(wiki_user.username)
+        )
+    )
+
+    document = create_document(save=True)
+    revision = create_revision(
+        title='My First And Only Revision',
+        document=document,
+        creator=wiki_user,
+        save=True)
+    assert Revision.objects.filter(creator=wiki_user).exists()
+
+    RevisionAkismetSubmission.objects.create(
+        revision=revision, sender=wiki_user)
+
+    attachment_revision = AttachmentRevision(
+        attachment=Attachment.objects.create(title='test attachment'),
+        file='some/path.ext',
+        mime_type='application/kuma',
+        creator=wiki_user,
+        title='test attachment',
+    )
+    attachment_revision.save()
+    assert AttachmentRevision.objects.filter(creator=wiki_user).exists()
+
+    url = reverse('users.user_delete', kwargs={'username': wiki_user.username})
+    response = user_client.post(url, {'attributions': 'donate'}, HTTP_HOST=settings.WIKI_HOST)
+    assert response.status_code == 302
+    assert not User.objects.filter(username=wiki_user.username).exists()
+
+    revision.refresh_from_db()
+    assert revision.creator.username == 'Anonymous'
+
+    attachment_revision.refresh_from_db()
+    assert attachment_revision.creator.username == 'Anonymous'
+
+    # The user_client should now become "invalid" since its session
+    # is going to point to no user.
+    response = user_client.get(url, HTTP_HOST=settings.WIKI_HOST)
+    assert response.status_code == 302
+    assert reverse('account_login') in response['Location']
+    # Let's doublecheck that
+    whoami_url = reverse('api.v1.whoami')
+    response = user_client.get(whoami_url)
+    assert response.status_code == 200
+    assert not response.json()['username']
+    assert not response.json()['is_authenticated']
+
+
+def test_delete_user_keep_attributions(db, user_client, wiki_user):
+
+    # Pretend the user logged in with GitHub
+    SocialAccount.objects.create(
+        user=wiki_user,
+        provider='github',
+        extra_data=dict(
+            email=wiki_user.email,
+            avatar_url='https://avatars0.githubusercontent.com/yada/yada',
+            html_url="https://github.com/{}".format(wiki_user.username)
+        )
+    )
```

Comment:
Same here, you can use the `wiki_user_github_account` pytest fixture instead of creating this yourself.